src/cryptenroll/cryptenroll.c

```diff
@@ -768,9 +768,15 @@ static int run(int argc, char *argv[]) {
                 break;
 
         case ENROLL_TPM2:
-                slot = enroll_tpm2(cd, vk, vks, arg_tpm2_device, arg_tpm2_seal_key_handle, arg_tpm2_device_key, arg_tpm2_hash_pcr_values, arg_tpm2_n_hash_pcr_values, arg_tpm2_public_key, arg_tpm2_public_key_pcr_mask, arg_tpm2_signature, arg_tpm2_pin, arg_tpm2_pcrlock);
-                break;
+                slot = enroll_tpm2(cd, vk, vks, arg_tpm2_device, arg_tpm2_seal_key_handle, arg_tpm2_device_key, arg_tpm2_hash_pcr_values, arg_tpm2_n_hash_pcr_values, arg_tpm2_public_key, arg_tpm2_public_key_pcr_mask, arg_tpm2_signature, arg_tpm2_pin, arg_tpm2_pcrlock, &slot_to_wipe);
 
+                if (slot >= 0 && slot_to_wipe >= 0) {
+                        /* Rotating PIN on an existing enrollment */
+                        r = wipe_slots(cd, &slot_to_wipe, 1, WIPE_EXPLICIT, 0, -1);
```

Comment:
according to our most recent coding style we generally want that parameters with very opaque values are prefixed with a comment explaining what they are. i.e. the third, fifth and sixth argument here are of that kind, you wouldn't have any idea what they might be if you just look at this line.

Please change the args to `/* n_explicit_slots= */ 1`, `/* by_mask= */ 0` and `/* except_slot= */ -1`, you get the idea